GitHub action build failed due to docker login issues

<!--
You don't need to remove this comment section, it's invisible on the issues page.

## General remarks

* Attention, please fill out this issues form using English only!
* 注意！GitHub Issue 仅支持英文，中文 Issue 请在 [论坛](https://kubesphere.com.cn/forum/) 提交。
* This form is to report bugs. For general usage questions you can join our Slack channel
       [KubeSphere-users](https://join.slack.com/t/kubesphere/shared_invite/enQtNTE3MDIxNzUxNzQ0LTZkNTdkYWNiYTVkMTM5ZThhODY1MjAyZmVlYWEwZmQ3ODQ1NmM1MGVkNWEzZTRhNzk0MzM5MmY4NDc3ZWVhMjE)
-->

**Describe the Bug**
GitHub action [build was failed](https://github.com/kubesphere/kubesphere/runs/1458657884). You can see the partial error logs below:
```
+ cat /home/runner/.docker/config.json
+ grep index.docker.io
		"https://index.docker.io/v1/": {
+ [[ 0 != 0 ]]
+ docker push kubespheredev/ks-apiserver:latest
The push refers to repository [docker.io/kubespheredev/ks-apiserver]
72a8d6ad82da: Preparing
0d737c3e8355: Preparing
3e207b409db3: Preparing
denied: requested access to the resource is denied
Error: Process completed with exit code 1.
```
It might be caused by #3129

**Versions Used**
None.

**Environment**
None.

**How To Reproduce**
None.

**Expected behavior**
The GitHub action build can be a success.

/kind bug

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -63,4 +63,6 @@ jobs:
           DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
           DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
         if: github.event_name == 'push'
-        run: bash hack/docker_build.sh ${{ steps.extract_branch.outputs.branch }}
+        run: |
+          echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+          bash hack/docker_build.sh ${{ steps.extract_branch.outputs.branch }}

